packages/tools/src/tools/annotation/planarFreehandROITool/interpolatePoints.ts

```diff
@@ -0,0 +1,176 @@
+import { Types } from '@cornerstonejs/core';
+import { point } from '../../../utilities/math';
+import interpolateSegmentPoints from './interpolation/interpolateSegmentPoints';
+
+export function assertInterpolation(configuration: Record<any, any>): boolean {
+  return configuration?.interpolation?.enabled === true;
+}
+
+/**
+ * Tells whether two points are equal by proximity or not as far as interpolation goes.
+ */
+function isEqualByProximity(pointA, pointB) {
+  return point.distanceToPoint(pointA, pointB) < 0.001;
+}
+
+/**
+ * Tells whether two points are strictly equal or not as far as interpolation goes.
+ */
+function isEqual(pointA, pointB) {
+  return point.distanceToPoint(pointA, pointB) === 0;
+}
+
+/**
+ * Finds the indexes of points list and otherPoints list that points are identical.
+ */
+function findMatchIndexes(
+  points: Types.Point2[],
+  otherPoints: Types.Point2[]
+): [number, number] | undefined {
+  for (let i = 0; i < points.length; i++) {
+    for (let j = 0; j < otherPoints.length; j++) {
+      if (isEqual(points[i], otherPoints[j])) {
+        return [i, j];
+      }
+    }
+  }
+}
+/**
+ * Returns the following index value (on circular basis) of index param on the given direction.
+ */
+function followingIndex(
+  index: number,
+  size: number,
+  direction: number
+): number {
+  return (index + size + direction) % size;
+}
+
+/**
+ * Circular finding that returns the next index for two list where the criteria is met.
+ */
+function circularFindNextIndexBy(
+  pointContent: [number, number, Types.Point2[]],
```

Comment:
This is too complex a type to really understand what it means from just the types - can you define it separately and have comments as to what the fields are, so that someone could re-use this function without having to necessarily trace through it?